Android_Development_with_Kotlin/09. Dynamic User Interface/Expandablelistview.md

```diff
@@ -0,0 +1,158 @@
+
+### ExpandingItem layout attributes
+
+Attribute Name         | Type         | Default Value | Meaning                                                             | Mandatory
+---------------------- | ------------ | ------------- | --------------------------------------------------------------------| --------
+item_layout            | reference    |               | The layout for the Item (header).                                   | Yes
+sub_item_layout        | reference    |               | The layout for the sub items.                                       | Yes
+separator_layout       | reference    |               | A layout to separate items.                                         | No
+indicator_size         | dimension    |       0dp     | The indicator size in dp.                                           | No
+indicator_margin_left  | dimension    |       0dp     | The margin between the indicator and its left.                      | No
+indicator_margin_right | dimension    |       0dp     | The margin between the indicator and its right.                     | No
+show_indicator         | boolean      |       true    | true if you want to show the indicator. false otherwise.            | No
+show_animation         | boolean      |       true    | true if you want to show animations. false otherwise.               | No
+start_collapsed        | boolean      |       true    | true if you want the sub views to start collapsed. false otherwise. | No
+animation_duration     | integer      |       300ms   | The animations duration in milliseconds.                            | No
+
+
+![expandablelistview](https://user-images.githubusercontent.com/70501362/135293881-5e129934-0543-43bc-9b82-8769de1b61d0.gif)
+
+
+### Step by Step Implementation
+
+**Step 1: Create a New Project**
+
+To create a new project in Android Studio please refer to [How to Create/Start a New Project in Android Studio](https://www.geeksforgeeks.org/android-how-to-create-start-a-new-project-in-android-studio/). Note that select **Kotlin** as the programming language.
+
+**Step 2: Working with the activity_main.xml file**
+
+Now go to the **activity_main.xml** file which represents the UI of the application. Below is the code for the **activity_main.xml** file. Comments are added inside the code to understand the code in more detail.
+
+## XML
+<?xml version="1.0"?>
+
+<img src="https://user-images.githubusercontent.com/78701779/134665601-f76558ef-b719-453d-82e0-c767f8831a50.png" height=550px width=475px>
+
+
+
+**Step 3: Working with the MainActivity file**
+
+Go to the **MainActivity** file, and refer to the following code. Below is the code for the both **MainActivity.kt** and **MainActivity.java **file. Comments are added inside the code to understand the code in more detail.
+
+## Kotlin
+
+<pre><code>
+// Kotlin code to implement the expandablelistview
+package com.journaldev.expandablelistview;
```

Comment:
change package name, to avoid copyright issues, and mention the site you referred at the bottom of this file